### PR TITLE
Add tests for direct editing and cleanup dead code

### PR DESCRIPTION
**Pull request by juliushaertl**
Remove the DirectSessionController which was never used and add some basic cypress tests for direct editing.

I wanted to check if the file has properly been saved, but that fails currently due to the fact that the save is not happening fast enough. Something to be looked in sepeartely though.

### CODE REVIEW

File: cypress/e2e/directediting.spec.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { initUserAndFiles, randHash } from '../utils/index.js'
+
+const randUser = randHash()
+const fileName = 'empty.md'
```
Comment (marcelklehr) [inline cypress/e2e/directediting.spec.js:4]: Can you clean up these lint errors?

**Review by marcelklehr — APPROVED** [review]

File: cypress/e2e/directediting.spec.js (hunk 1 above)
Comment (juliushaertl) [reply cypress/e2e/directediting.spec.js]: Done

**Comment by juliushaertl** [comment]
Follow up issue for the fixme comment and saving problem: https://github.com/nextcloud/text/issues/3404

**Review by mejo- — APPROVED** [review]
Not sure about the failing `mobile.spec.js` Cypress tests, but they seem unrelated to this PR.

**Pull request merged**